brave/brave-ios
# Fix 8064: Some v2 updates in NFT details

## Summary of Changes


This pull request fixes #8064

## Submitter Checklist:


## Test Plan:
1. Set up wallet and make sure add or auto-discover some NFTs
2. Click any NFT to go to its details screen 
3. The NFT image now has a network icon and an account blockie that owns this NFT (This is missing when you come from asset search list. Not listed in the figma, if this is required we can add later)
4. On top right there is a 3-dot button which will give you the same options when you long-press the NFT grid in the previous screen + a new option `Send` if this NFT is visible 
5. Click send will bring you to the send screen
6. Click hide will update NFT visible status which will change the options of the 3-dot, and move the NFT to the `Hide` selection if you go back to the previous screen
7. Click unhide will update NFT visible status which will change the options of the 3-dot, and move the NFT to the `Collected` selection if you go back to the previous screen.
8. Click mark as not junk will update NFT spam status which will change the options of the 3-dot, and move the NFT to the `Collected` selection if you go back to the previous screen
9. Click Don't show in wallet will remove NFT from the wallet after you confirm to remove it. You will be pop-off from the details screen and NFT will be remove from the wallet entirely. 
10. Below the image, we have `Overview`, `Description` and `Properties` sections(optional), if we have information from NFT's metadata. 

## Screenshots:


![Simulator Screenshot - iPhone 15 Pro - 2023-11-14 at 17 47 09](https://github.com/brave/brave-ios/assets/1187676/99988829-19ea-4932-ae40-083a2f700313) | ![Simulator Screenshot - iPhone 15 Pro - 2023-11-14 at 17 47 44](https://github.com/brave/brave-ios/assets/1187676/30ed8a83-50e1-4078-a8e3-35a4adc5b245)
--- | ---
![Simulator Screenshot - iPhone 15 Pro - 2023-11-14 at 17 47 13](https://github.com/brave/brave-ios/assets/1187676/e416ff2d-7a4d-4a08-a418-9ace7842960d) | ![Simulator Screenshot - iPhone 15 Pro - 2023-11-14 at 17 47 47](https://github.com/brave/brave-ios/assets/1187676/0fd870da-568b-4a70-afa8-28b635f5bd29)
![Simulator Screenshot - iPhone 15 Pro - 2023-11-14 at 17 47 19](https://github.com/brave/brave-ios/assets/1187676/f959f8cb-4f14-4936-92a6-5e88c33edcf5) | ![Simulator Screenshot - iPhone 15 Pro - 2023-11-14 at 17 47 41](https://github.com/brave/brave-ios/assets/1187676/4c4829dd-ee95-4311-8393-65cc5e897afa)
![Simulator Screenshot - iPhone 15 Pro - 2023-11-14 at 17 48 49](https://github.com/brave/brave-ios/assets/1187676/1cd38e94-238f-46e0-8452-e01f55a10bfd) | ![Simulator Screenshot - iPhone 15 Pro - 2023-11-14 at 21 09 57](https://github.com/brave/brave-ios/assets/1187676/76440802-4bc7-492e-84ad-5567f2ee2d47)

## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
